wordle/src/main/kotlin/domain/Word.kt

```diff
@@ -0,0 +1,86 @@
+package domain
+
+import domain.Tile.*
+
+private const val WORD_LENGTH = 5
+
+class Word(
+    word: String
+) {
+    val letters: List<Letter>
+
+    init {
+        if (word.length != WORD_LENGTH) {
+            throw IllegalArgumentException("글자수는 반드시 5글자여야 합니다. (입력: ${word.length}글자)")
+        }
+
+        letters = word.map { letter -> Letter(letter.toString()) }.toList()
+    }
+
+    fun compareWithCorrectAnswer(correctAnswer: Word): Tiles {
+        val answerLetters = ArrayList<Letter?>(correctAnswer.letters)
+
+        var tiles = listOf(GRAY, GRAY, GRAY, GRAY, GRAY)
+        tiles = judgeGreen(tiles, answerLetters)
+        tiles = judgeYellow(tiles, answerLetters)
+
+        return Tiles(tiles)
+    }
+
+    private fun judgeGreen(
+        tiles: List<Tile>,
+        answerLetters: ArrayList<Letter?>
+    ): List<Tile> {
+        return tiles.mapIndexed { i, tile ->
+            if (letters[i] == answerLetters[i]) {
+                GREEN
+            } else {
+                tile
+            }
```

Comment:
```suggestion
            if (letters[i] == answerLetters[i]) GREEN else tile
```
이런 형태는 어떤가요! 저는 몇번 써보니 마음에 들더라고요

Reply:
저도 코틀린은 이렇게 간결하게 쓰는게 더 좋다는 생각이 요즘 많이 들어요 👍👍